src/v/kafka/server/quota_manager.cc

```diff
@@ -252,50 +252,47 @@ quota_manager::add_quota_id(tracker_key qid, clock::time_point now) {
-    // vassert(
-    //   ss::this_shard_id() == _client_quotas.shard_id(),
-    //   "update_client_quotas must only be called on the owner shard");
-
-    // ssx::spawn_with_gate(_gate, [this] {
-    //     return _client_quotas.update([this](client_quotas_map_t quotas) {
-    //         constexpr auto set_bucket =
-    //           [](
-    //             std::optional<atomic_token_bucket>& bucket,
-    //             std::optional<uint64_t> rate,
-    //             std::optional<uint64_t> replenish_threshold) {
-    //               if (!rate) {
-    //                   bucket.reset();
-    //                   return;
-    //               }
-    //               if (bucket.has_value() && bucket->rate() == rate) {
-    //                   return;
-    //               }
-    //               bucket.emplace(
-    //                 *rate, *rate, replenish_threshold.value_or(1), true);
-    //           };
-
-    //         for (auto& quota : quotas) {
-    //             auto limits = _translator.find_quota_value(quota.first);
-    //             set_bucket(
-    //               quota.second->tp_produce_rate,
-    //               limits.produce_limit,
-    //               _replenish_threshold());
-
-    //             set_bucket(
-    //               quota.second->tp_fetch_rate,
-    //               limits.fetch_limit,
-    //               _replenish_threshold());
-
-    //             set_bucket(
-    //               quota.second->pm_rate,
-    //               limits.partition_mutation_limit,
-    //               _replenish_threshold());
-    //         }
-
-    //         return quotas;
-    //     });
-    // });
+    vassert(
+      ss::this_shard_id() == quotas_shard,
+      "update_client_quotas must only be called on the owner shard");
+
+    ssx::spawn_with_gate(_gate, [this] {
+        constexpr auto set_bucket =
+          [](
+            std::optional<atomic_token_bucket>& bucket,
+            std::optional<uint64_t> rate,
+            std::optional<uint64_t> replenish_threshold) {
+              if (!rate) {
+                  bucket.reset();
+                  return;
+              }
+
+              if (bucket.has_value() && bucket->rate() == rate) {
+                  return;
+              }
+              bucket.emplace(
+                *rate, *rate, replenish_threshold.value_or(1), true);
+              return;
+          };
+
+        for (auto& quota : *_global_map) {
+            auto limits = _translator.find_quota_value(quota.first);
+            set_bucket(
+              quota.second->tp_produce_rate,
+              limits.produce_limit,
+              _replenish_threshold());
+
+            set_bucket(
+              quota.second->tp_fetch_rate,
+              limits.fetch_limit,
+              _replenish_threshold());
+
+            set_bucket(
+              quota.second->pm_rate,
+              limits.partition_mutation_limit,
+              _replenish_threshold());
+        }
+    });
```

Comment:
Can this change go in the previous commit?

Reply:
Yep. I've reworked the commits now to reduce code motion and make the diffs easier to review.